Tryouts/Plugins/ApplicationPlugins/process explorer/backend/ModuleProcessMonitor/Processes/ProcessInfoGeneratorWindows.cs

```diff
@@ -0,0 +1,360 @@
+﻿// Morgan Stanley makes this available to you under the Apache License,
+// Version 2.0 (the "License"). You may obtain a copy of the License at
+// 
+//      http://www.apache.org/licenses/LICENSE-2.0.
+// 
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership. Unless required by applicable law or agreed
+// to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions
+// and limitations under the License.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Management;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using ModuleProcessMonitor.Logging;
+
+namespace ModuleProcessMonitor.Processes;
+
+[System.Runtime.Versioning.SupportedOSPlatform("windows")]
+internal class ProcessInfoGeneratorWindows : ProcessInfoManager
```

Comment:
Consider implementing IDisposable to handle shutting down without handling WMI events